### PR TITLE
Fix certificate expired notification every second

### PR DESCRIPTION
**Pull request by fantostisch**
Fixes https://github.com/eduvpn/android/issues/329 caused by 942876402d3f1d312ef6304cd12c4cd366a38956.

### CODE REVIEW

File: app/src/main/java/nl/eduvpn/app/viewmodel/ConnectionStatusViewModel.kt [hunk 1]
```diff
@@ -123,7 +123,8 @@ class ConnectionStatusViewModel @Inject constructor(
             certValidity.value = HtmlCompat.fromHtml(context.getString(R.string.connection_certificate_status_expired), HtmlCompat.FROM_HTML_MODE_COMPACT)
             _connectionParentAction.value = ParentAction.SessionExpired
             return false
-        } else if (timeDifferenceInSeconds < 60) {
+        }
+        if (timeDifferenceInSeconds < 60) {
```
Comment (dzolnai) [inline app/src/main/java/nl/eduvpn/app/viewmodel/ConnectionStatusViewModel.kt:127]: Could you explain how this fixes the issue?
Isn't the problem that updateCertExpiry() will get called every second with negative time difference values?

Comment (fantostisch) [reply app/src/main/java/nl/eduvpn/app/viewmodel/ConnectionStatusViewModel.kt:127]: Commit cd540c1440c46d1c759dc3b299a6f69d2ea4bc2e does not change how the code works, only how it looks. The problem was that `runUpdateCertExpiryEverySecond` was called after `stopUpdateCertExpiry`. So after the `updateCertCallback` was removed in https://github.com/eduvpn/android/blob/599a22e0a3890236068cfc3a4eadda0212a3d662/app/src/main/java/nl/eduvpn/app/viewmodel/ConnectionStatusViewModel.kt#L114 or https://github.com/eduvpn/android/blob/599a22e0a3890236068cfc3a4eadda0212a3d662/app/src/main/java/nl/eduvpn/app/viewmodel/ConnectionStatusViewModel.kt#L121 it was added again in https://github.com/eduvpn/android/blob/599a22e0a3890236068cfc3a4eadda0212a3d662/app/src/main/java/nl/eduvpn/app/viewmodel/ConnectionStatusViewModel.kt#L64 The callback is now not added anymore when `updateCertExpiry` returns false.

**Comment by fantostisch** [comment]
There is still a bug, `stopUpdateCertExpiry` will not work when `updateCertCallback` is running.

**Comment by dzolnai** [comment]
> There is still a bug, `stopUpdateCertExpiry` will not work when `updateCertCallback` is running.

Will you update the pull later with an extra commit to fix this?

**Comment by fantostisch** [comment]
I just tested (https://github.com/fantostisch/android/commit/2de2707031539d14e49a81f9de89709e2cd36c25) to be sure and somehow `onPause` will not be called before `updateCertExpiry` is finished, so the bug I thought was there is not there. While testing I discovered that the code waits 1 second before it starts calculating the cert expiry. This is fixed in https://github.com/eduvpn/android/pull/331/commits/e4fa5b1634aa3ac74e85dbb99accfd736754db83.

**Review by dzolnai — APPROVED** [review]

**Comment by dzolnai** [comment]
From my part, yes.

**Comment by fkooman** [comment]
Thanks!

**Pull request merged**